src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -106,37 +108,24 @@ private class LatestDiagnosticsForSpanGetter
 
             public async Task<bool> TryGetAsync(List<DiagnosticData> list, CancellationToken cancellationToken)
             {
+                var containsFullResult = true;
+                var loggingEnabled = RoslynEventSource.Instance.IsEnabled(EventLevel.Informational, EventKeywords.None);
+
                 try
                 {
-                    var containsFullResult = true;
                     foreach (var stateSet in _stateSets)
                     {
-                        cancellationToken.ThrowIfCancellationRequested();
-
-                        containsFullResult &= await TryGetSyntaxAndSemanticDiagnosticsAsync(stateSet, list, cancellationToken).ConfigureAwait(false);
-
-                        // check whether compilation end code fix is enabled
-                        if (!_document.Project.Solution.Workspace.Options.GetOption(InternalDiagnosticsOptions.CompilationEndCodeFix))
-                        {
-                            continue;
-                        }
-
-                        // check whether heuristic is enabled
-                        if (_blockForData && _document.Project.Solution.Workspace.Options.GetOption(InternalDiagnosticsOptions.UseCompilationEndCodeFixHeuristic))
+                        if (loggingEnabled)
```

Comment:
❕ Don't use conditional logic here. The code path should not be different when instrumentation is enabled. We also need the ending event to be logged if recording starts in the middle of a long-running operation, and we want the ending event to not be reported if recording stops in the middle of an operation.